src/kernel/capinv.c

```diff
@@ -569,9 +577,8 @@ capinv_int_snd(struct thread *rcv_thd, struct pt_regs *regs)
 	assert(rcv_tcap);
 
 	next = asnd_process(rcv_thd, thd, rcv_tcap, tcap);
-	if (next == thd) return 0;
+	if (next == thd) return 1; /* current thread is a preempted one! */
```

Comment:
I don't know if the return value of the function is checked anywhere, so I'm not sure if it matters either way.